experimental/torch_xla2/torch_xla2/ops/jaten.py

```diff
@@ -4434,3 +4434,86 @@ def _aten_upsample_bilinear2d_aa(input, output_size, align_corners, scale_factor
         translation=translation,
         antialias=antialias,
     )
+
+@op(torch.ops.aten.cdist)
+def _aten_cdist(x1, x2, p=2.0, compute_mode='use_mm_for_euclid_dist_if_necessary'):
+  x1 = x1.astype(jnp.float32)
+  x2 = x2.astype(jnp.float32)
+
+  if p == 0.0:
+    # For p = 0, use Hamming-like distance multiplied by the number of elements
+    return _hamming_distance(x1, x2).astype(jnp.float32)
+  elif p == 2.0:
+    # Use optimized Euclidean distance calculation
+    if compute_mode == 'use_mm_for_euclid_dist_if_necessary' and (x1.shape[-2] > 25 or x2.shape[-2] > 25):
+      return _euclidean_mm(x1, x2)
+    elif compute_mode == 'use_mm_for_euclid_dist':
+      return _euclidean_mm(x1, x2)
+    else:
+      return _euclidean_direct(x1, x2)
+  else:
+    # General p-norm distance calculation
+    diff = jnp.abs(jnp.expand_dims(x1, -2) - jnp.expand_dims(x2, -3))
+    return jnp.sum(jnp.power(diff, p), axis=-1).astype(jnp.float32) ** (1 / p)
+
```

Comment:
Thanks for the contribution, do we have any reference for others need to know why we need different implementation for these specific difference situtions?